src/main/java/org/broadinstitute/consent/http/authentication/OAuthAuthenticator.java

```diff
@@ -102,9 +99,15 @@ private AuthUser getUserWithStatusInfo(AuthUser authUser) {
         } else {
           logWarn("Error posting to Sam, AuthUser not able to be registered: " + gson.toJson(authUser));
         }
-      } catch (Exception exc) {
-        logException("AuthUser not able to be registered: '" + gson.toJson(authUser), exc);
+        // if post response is not successful this will be thrown, propagate the error to the user
+      } catch (WebApplicationException ex) {
+        throw ex;
+        // if building the request or parsing the response fails, this will be thrown
+        // authenticationExceptions are caught and rethrown as a 500 error in AuthFilter
+      } catch (Exception ex2) {
+        throw new AuthenticationException("AuthUser not able to be registered: '" + gson.toJson(authUser), ex2);
```

Comment:
In this scenario, the AuthUser HAS authenticated successfully, so it doesn't make sense to me that we throw an `AuthenticationException`. The error is a Sam specific registration error. Can we create a new exception case for this, maybe something like `RegistrationException extends WebApplicationException` or something similar? Open to other suggestions on this.

Reply:
I think that could be a good idea- I am open to changing this because AuthenticationExceptions get caught and rethrown as general 500 errors. But, just to clarify - Has the user authed successfully in this case? If getting the registration info fails we try to post the registration info, and this is the case that posting also failed either because the request couldn't be built or the response couldn't be parsed.

Reply:
> Has the user authed successfully in this case? If getting the registration info fails we try to post the registration info, and this is the case that posting also failed either because the request couldn't be built or the response couldn't be parsed. I guess what is the difference between failing to register a user and failing to auth?

An authed user has gone through the authentication dance with the selected identity provider, Google or MS. We have a valid auth token and we know there is a real identity. A registered Sam user is an identity that Sam knows about and has information about - it's a completely separate entity.